tests/conftest.py

```diff
@@ -202,11 +199,21 @@ def applySignature(self, signatureName, address):
         return contract
 
     def resetSnapshot(self):
-        self.chain.block = self.originalBlock
-        self.chain.revert(self.snapshot)
-        self.chain.head_state = self.originalHead
-        self.contracts = deepcopy(self.originalContracts)
+        self.resetToSnapshot(self.captured)
 
+    def createSnapshot(self):
+        self.chain.mine(1)
+        self.captured = {'block': self.chain.block, 'head_state': self.chain.head_state, 'contracts' : deepcopy(self.contracts), 'snapshot': self.chain.snapshot()}
```

Comment:
Don't assign to `self.captured` here.  Instead just return this dictionary and then up on line 162 do: `self.captured = self.createSnapshot()`.  This also means you don't need to do line 161 (`self.captured = {}`).

Reply:
doh! makes sense.

Reply:
The reason for this is that various fixtures will create different snapshots and so there may be many snapshots floating around in the wild at a time, not just one.